src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1368,6 +1427,54 @@ cpp2::ErrorCode RaftPart::verifyLeader(
     return cpp2::ErrorCode::SUCCEEDED;
 }
 
+void RaftPart::processSendSnapshotRequest(const cpp2::SendSnapshotRequest& req,
+                                          cpp2::SendSnapshotResponse& resp) {
+    LOG(INFO) << idStr_ << "Receive snapshot, total rows " << req.get_rows().size()
+              << ", total count received " << req.get_total_count()
+              << ", total size received " << req.get_total_size()
+              << ", finished " << req.get_done();
+    std::lock_guard<std::mutex> g(raftLock_);
+    // Check status
+    if (UNLIKELY(status_ == Status::STOPPED)) {
+        VLOG(2) << idStr_
+                << "The part has been stopped, skip the request";
+        resp.set_error_code(cpp2::ErrorCode::E_BAD_STATE);
+        return;
+    }
+    if (UNLIKELY(status_ == Status::STARTING)) {
+        VLOG(2) << idStr_ << "The partition is still starting";
+        resp.set_error_code(cpp2::ErrorCode::E_NOT_READY);
+        return;
+    }
+    if (status_ != Status::WAITING_SNAPSHOT) {
+        LOG(INFO) << idStr_ << "Begin to receive the snapshot";
+        wal_->reset();
+        cleanup();
+        lastLogId_ = committedLogId_ = 0;
+        lastTotalCount_ = 0;
+        lastTotalSize_ = 0;
```

Comment:
code follow exist three place, do it in a inner function?
wal_->reset();
        cleanup();
        lastLogId_ = committedLogId_ = 0;
        lastTotalCount_ = 0;
        lastTotalSize_ = 0;

Reply:
Good point.